designs/component-upgrades.md

```diff
@@ -0,0 +1,149 @@
+# EKS-A components upgrade
+
+
+## Introduction
+
+**Problem:** customers are only able to upgrade the cluster components that are part of the kubernetes distribution. Any other core component of EKS-A remains untouched from the cluster creation.
+
+This stops customers from enjoying the latest features and bug fixes from eks-a and any of its core components as well as from being able to get security patches.
+
+### Tenets
+
+* **Simple:** simple to use, simple to maintain, simple to build
+
+### Goals and objectives
+
+As an eks-a customer:
+
+* I want to able to enjoy the latest eks-a features and bug fixes
+* I want to keep my cluster up to date with security patches
+* I don’t want to need to stay informed about latest versions, security patches or compatibility between components
+
+### Statement of scope
+
+**In scope**
+
+* Upgrade core eks-a components to latest versions available for an specific cli version
+    * Core CAPI
+    * CAPI providers
+    * Etcdadm CAPI provider
+    * CNIs (cilium)
+    * EKS-A controller and CRDs
+    * Flux
+
+**Not in scope**
+
+* Install/upgrade add-ons
```

Comment:
This will be in scope whenever we have add ons that are part of the create workflow right? Unless we have documentation mentioning otherwise.

Reply:
I'm not sure we have such documentation
However, I was thinking that add-ons might have their own lifecycle system, whether they are upgraded through the `upgrade cluster` command or not. So I considered their upgrade flow a different feature, so not in scope for this one.

Reply:
So is Flux the only exception here?

Reply:
Yeah Flux is the exception, it's not a critical component but it's a core component of the eks-a experience